### PR TITLE
Issue #923 Support eviction veto in XA caches

### PR DESCRIPTION
**Pull request by ljacomet**

### CODE REVIEW

File: transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java [hunk 1]
```diff
@@ -979,4 +974,22 @@ public void stop() {
...
+
+    private final EvictionVeto<? super K, ? super V> wrappedVeto;
+
+    private XAEvictionVeto(EvictionVeto<? super K, ? super V> wrappedVeto) {
+      this.wrappedVeto = wrappedVeto;
+    }
+
+    @Override
+    public boolean vetoes(K key, SoftLock<V> softLock) {
+      if (isInDoubt(softLock)) {
```
Comment (chrisdennis) [inline transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java]:
```
return isInDoubt(softLock) || wrappedVeto.vetoes(key, softLock.getOldValue());
```

Comment (ljacomet) [inline transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java]: My IDE agrees with you ... still find the expanded version more readable.
Others to chime in?

Comment (GaryWKeim) [inline transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java]: I prefer conciseness.

Comment (ljacomet) [inline transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java]: Changing.

File: transactions/src/test/java/org/ehcache/transactions/xa/internal/XAStoreTest.java [hunk 2]
```diff
@@ -1488,6 +1490,52 @@ public void testBulkComputeIfAbsent() throws Exception {
   }
 
   @Test
+  public void testCustomEvictionVeto() throws Exception {
```
Comment (chrisdennis) [inline transactions/src/test/java/org/ehcache/transactions/xa/internal/XAStoreTest.java]: Are we ruling this test sufficient?  It obviously verifies that the custom veto is called... it doesn't actually verify that there is any effect on the eviction process however (of course whether that could be verified is another question).

Comment (ljacomet) [inline transactions/src/test/java/org/ehcache/transactions/xa/internal/XAStoreTest.java]: The only thing that needs verification at this level is that the provided veto is called. Testing the proper handling of veto exists elsewhere.

**Pull request merged**